salt/modules/reg.py

```diff
@@ -555,42 +490,49 @@ def set_value(hive,
         salt '*' reg.set_value HKEY_LOCAL_MACHINE 'SOFTWARE\\Salt' 'version' '2015.5.2' \\
         vtype=REG_LIST vdata='[a,b,c]'
     '''
-
-    if PY2:
```

Comment:
From a performance point of view test IF something is PY2  vs an IF for every string inside _to_unicode() i.e. one IF vs many IFs   When PY2 is dropped it makes it easy to pull the old code out.  As the conversion to Unicode will no loner be needed.

Reply:
The conversion will be removed in `salt.utils.to_unicode` when we go away from Py2

Reply:
I would have thought that the function salt.utils.to_unicode from a PY3 point of view would convert bytes to unicode.    salt.utils.to_unicode  this does not need to be called in PY3 at all inside reg.py.  Why then call a function which does 0 but has an overhead..  I would have thought dropping PY2 support for windows is less  18 months away because PY2 is not up to the job. (might be wishful thinking)

Reply:
It makes the code easier to maintain.